### ISSUE TITLE
HierarchicalMNS cannot parse easyconfig

### ISSUE THREAD

**Issue by ocaisa** [hpcugent/easybuild-framework#1402]
For some reason, `binutils-2.25-GCC-4.9.3-binutils-2.25.eb` cannot be parsed when using `HierarchicalMNS`:
```
$ eb  binutils-2.25-GCC-4.9.3-binutils-2.25.eb  --dry-run --module-naming-scheme=HierarchicalMNS
== temporary log file in case of crash /tmp/eb-IBa7go/easybuild-7uT1Np.log
ERROR: Failed to process easyconfig /home/alanc/EasyBuild_Git/easybuild-easyconfigs/easybuild/easyconfigs/b/binutils/binutils-2.25-GCC-4.9.3-binutils-2.25.eb: Failed to find easyconfig file 'flex-2.5.39-GCC-4.9.3-binutils-%(version)s.eb' when determining module name for: {'external_module_metadata': {}, 'dummy': False, 'toolchain': {'version': '4.9.3-binutils-%(version)s', 'name': 'GCC'}, 'name': 'flex', 'full_mod_name': None, 'short_mod_name': None, 'versionsuffix': '', 'version': '2.5.39', 'hidden': False, 'external_module': False}
```

Ideas @geimer ?

**Comment by ocaisa** [comment]
Is this an easyconfig problem or a parser? It seems to be trivially fixed by replacing
```
toolchain = {'name': 'GCC', 'version': '4.9.3-binutils-%(version)s' }
```
with
```
toolchain = {'name': 'GCC', 'version': '4.9.3-binutils-%s' % version}
```

**Comment by boegel** [comment]
@ocaisa: this indicates a problem in the framework somewhere; the `%(version)s` template is not being resolved

seems like a chicken-egg situation (the version is only available after parsing the easyconfig)

**Comment by boegel** [comment]
I'll merge https://github.com/hpcugent/easybuild-easyconfigs/pull/2104 as a workaround for this until we can figure out a proper fix.

**Comment by boegel** [comment]
Seems like this problem is more extensive than I realised, see https://github.com/hpcugent/easybuild-easyconfigs/issues/3210 reported by @verdurin... I'll try and fix this by the next release.

**Comment by boegel** [comment]
This is fixed with #1812, @ocaisa: please verify?

**Comment by boegel** [comment]
fixed with #1812
